schutzbot/get_civ_config.py

```diff
@@ -1,119 +1,136 @@
 """
-This script exports checks which files and test methods have changed. If the only thing
-that changed in the PR are test methods, only execute those.
-
-To do so, it creates bash script with "SKIP_<CLOUD>" variables and "CIV_CONFIG_FILE" that
-is a file with the configuration of CIV in yaml format.
+Advanced CI test selection logic for cloud-image-val:
+- If ONLY specific test methods in a cloud test file are changed, run only those methods for that cloud.
+- If ONLY specific test methods in a generic test file are changed, run only those methods for all clouds.
+- If any other files or logic are changed (including fixtures, helpers, or CI scripts), run all tests for all clouds.
 """
+
 import os
+import re
 import subprocess
 import sys
 import yaml
-
 from pprint import pprint
 
+# Define test file locations
+CLOUD_TEST_FILES = {
+    'aws': 'test_suite/cloud/test_aws.py',
+    'azure': 'test_suite/cloud/test_azure.py',
+    'gcp': 'test_suite/cloud/test_gcp.py',
+}
+GENERIC_TEST_FILE = 'test_suite/generic/test_generic.py'
+
 
 def get_files_changed():
-    os.system('git remote add upstream https://github.com/osbuild/cloud-image-val.git')
-    os.system('git fetch upstream')
+    subprocess.run('git remote add upstream https://github.com/osbuild/cloud-image-val.git', shell=True, check=True)
```

Comment:
Not sure why you would need "shell=True" in line 25.
As the input is hardcoded it probably doesn't matter but it could still resolve unwanted ENVs at runtime.